src/crafting.h

```diff
@@ -104,6 +104,84 @@ struct recipe {
 
 };
 
+/**
+*   enum used by comp_selection to indicate where a component should be consumed from.
+*/
+enum usage {
+    use_from_map = 1,
+    use_from_player = 2,
+    use_from_both = 1 | 2,
+    use_from_none = 4,
+    cancel = 5 // FIXME: hacky.
```

Comment:
> Does defining the enum this way tell the compiler it can treat it's values differently than when defining just  3 ?

No, values are values. Combining them via bitwise or has the advantage that you can test it easier: `(value & use_from_map)` instead of `(value == use_from_map || value == use_from_both)`.

Btw. It should have been written `use_from_both = use_from_map | use_from_player` instead of the magic values, but that's the same to the compiler.